### ISSUE TITLE
DecrementInteger is not mutating with big integers

### ISSUE THREAD

**Issue by kubawerlos** [infection/infection#1661]
| Question    | Answer
| ------------| ---------------
| Infection version | 0.25.3 (tested on playground)
| Test Framework version | PHPUnit version: 9.5.9
| PHP version | Add `verbose` in playground and I will tell you :)
| Platform    | Ubuntu I guess
| ~Github Repo~ Example | https://infection-php.dev/r/78r7

I saw there is a test for comparing to [`0`](https://github.com/infection/infection/blob/0.26.4/tests/phpunit/Mutator/Number/DecrementIntegerTest.php#L148) and `1`, but I'd expect the mutant when comparing to `42` (or actually anything greater than `1`), or the behaviour is correct (then test case would be useful to make it explicit).

**Comment by maks-rafalko** [comment]
Thanks, feel free with helping fixing it :)

**Comment by LeoVie** [comment]
Hmm, this seems to be pretty much intentional, doesn't it?
In git history, I've found https://github.com/infection/infection/pull/639 which introduces the test case "It does not decrement an integer in a comparison".

> Since changing the number by one is basically the same as changing the comparison operator.

If I'm right, this issue could be closed? @maks-rafalko

**Comment by kubawerlos** [comment]
> Since changing the number by one is basically the same as changing the comparison operator.

Not really, it is not. Take a look at the [example](https://infection-php.dev/r/78r7) - this clearly lacks the border values, how would you change the operator there?

**Comment by staabm** [comment]
I think you are looking for the wrong mutator.
Its covered by the GreaterThan mutator

https://infection-php.dev/r/23vn

**Comment by maks-rafalko** [comment]
Agree, thank you all for the clarifications.

`DecrementInteger` does not mutate here as this is covered by `GreaterThan` and similar. 

Feel free to reopen if you are not agree.
